src/intrinsics/native/divmod.rs

```diff
@@ -164,9 +164,15 @@ pub fn udivmod4(
 
 // See Knuth, TAOCP, Volume 2, section 4.3.1, Algorithm D.
 // https://skanthak.homepage.t-online.de/division.html
+// The high word of v (the divisor) must be non-zero.
 #[inline]
 pub fn div_mod_knuth(u: &U256, v: &U256) -> (U256, U256) {
     const N_UDWORD_BITS: u32 = 128;
+    assert_ne!(
+        *u.high(),
+        0,
+        "The second operand must be greater than u128::MAX"
+    );
```

Comment:
nit: Unrelated to this PR, but this should probably not be `pub`, as it is intended to be called only when dividing by large dividends. This would allow this assertion to be `debug_assert`:

```suggestion
fn div_mod_knuth(u: &U256, v: &U256) -> (U256, U256) {
    const N_UDWORD_BITS: u32 = 128;
    debug_assert_ne!(
        *u.high(),
        0,
        "The second operand must be greater than u128::MAX"
    );
```

Reply:
Making it private makes sense, but a debug assert still isn't enough. Assertions have the benefit of assuring the compiler that the assertion is true if the program hasn't already panicked. Debug assertions are assertions when debug assertions is on but do nothing when debug assertions are off. I think I can get the intended effect by using another unsafe compiler hint and adding `unsafe` to the function signature.